src/main/java/no/unit/nva/doi/FetchDoiMetadata.java

```diff
@@ -69,32 +79,47 @@ public GatewayResponse handleRequest(Map<String, Object> input, Context context)
         }
 
         try {
-            String doiMetadata = lookupDoiMetadata(doiLookup.getDoiUrl(), dataciteContentType);
-            return new GatewayResponse(doiMetadata, OK.getStatusCode(), dataciteContentType.getContentType());
+            String doiMetadata = lookupDoiMetadata(doiLookup.getDoi(), dataciteContentType);
+            return new GatewayResponse(doiMetadata, OK.getStatusCode(),
+                                       dataciteContentType.getContentType());
         } catch (IOException e) {
             System.out.println(e.getMessage());
+            logger.log(e.getMessage());
             return errorGatewayResponse(e.getMessage(), SERVICE_UNAVAILABLE.getStatusCode());
         } catch (Exception e) {
             System.out.println(e.getMessage());
+            logger.log(e.getMessage());
             return errorGatewayResponse(e.getMessage(), INTERNAL_SERVER_ERROR.getStatusCode());
 
         }
     }
 
-    private String lookupDoiMetadata(URL doiUrl, DataciteContentType dataciteContentType) throws IOException {
+    private void init(Context context) {
+        this.logger = context.getLogger();
+        this.crossRefClient.setLogger(logger);
+    }
+
+    private String lookupDoiMetadata(String doiUrl, DataciteContentType dataciteContentType)
+        throws IOException, InterruptedException, ExecutionException, URISyntaxException {
         System.out.println("getDoiMetadata(doi:" + doiUrl + ")");
-        return dataciteClient.fetchMetadata(doiUrl.toString(), dataciteContentType);
+        Optional<String> crossRefResult = crossRefClient.fetchDataForDoi(doiUrl);
+        if (crossRefResult.isEmpty()) {
```

Comment:
tried to chain them, but had problems with the exception handling. I have a suggestion for this that will allow us to have a more fluent syntax in the future